server/src/main/java/org/elasticsearch/index/codec/PerFieldMapperCodec.java

```diff
@@ -84,6 +93,30 @@ public KnnVectorsFormat getKnnVectorsFormatForField(String field) {
 
     @Override
     public DocValuesFormat getDocValuesFormatForField(String field) {
+        if (useTSDBDocValuesFormat(field)) {
+            return tsdbDocValuesFormat;
+        }
         return docValuesFormat;
     }
+
+    private boolean useTSDBDocValuesFormat(final String field) {
+        return isTimeSeriesModeIndex() && isES97OrAbove() && (isTimestampField(field) || isTimeSeriesCounter(field));
```

Comment:
You don't need the `isES97OrAbove`: `PerFieldDocValuesFormat` records the format used by each field as part of the segment, so old indexes will not even try to use the new doc values format.